vertical-pod-autoscaler/pkg/recommender/Dockerfile

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM --platform=$BUILDPLATFORM golang:1.22.4 as builder
+FROM --platform=$BUILDPLATFORM golang:1.23rc1 as builder
```

Comment:
Go 1.23 has not yet been released, dependabot updated golang version to 1.23rc1 which is a release candidate, not the actual release. Go 1.23 is expected to be released in August 2024.

IMO we can' update it to go1.23rc1

Reply:
Hi @voelzmo @kwiesmueller 
your thoughts on this.

Reply:
Closing this PR as dependabot opened PR #7007 to update the Golang version to 1.22.5

/close